src/matroskavideo.cpp

```diff
@@ -0,0 +1,763 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, 5th Floor, Boston, MA 02110-1301 USA.
+ */
+// *****************************************************************************
+// included header files
+#include "config.h"
+
+//#ifdef EXV_ENABLE_VIDEO
+#include "matroskavideo.hpp"
+#include "error.hpp"
+#include "futils.hpp"
+#include "basicio.hpp"
+#include "tags.hpp"
+#include "tags_int.hpp"
+
+// + standard includes
+#include <cmath>
+#include <cassert>
+#include<iostream>
+
+// *****************************************************************************
+// class member definitions
+namespace Exiv2 {
+    namespace Internal {
+
+    //! List of composite tags. They are skipped and the child tags are read immediately
+    uint64_t compositeTagsList[] = {
+        0x0000, 0x000e, 0x000f, 0x0020, 0x0026, 0x002e, 0x0036,
+        0x0037, 0x003b, 0x005b, 0x0060, 0x0061, 0x0068, 0x05b9,
+        0x0dbb, 0x1034, 0x1035, 0x1854, 0x21a7, 0x2240, 0x23c0,
+        0x2624, 0x27c8, 0x2911, 0x2924, 0x2944, 0x2d80, 0x3373,
+        0x35a1, 0x3e5b, 0x3e7b,
+        0x14d9b74, 0x254c367, 0x549a966, 0x654ae6b, 0x8538067,
+        0x941a469, 0xa45dfa3, 0xb538667, 0xc53bb6b, 0xf43b675
+    };
+
+    //! List of tags which are ignored, i.e., tag and value won't be read
+    uint64_t ignoredTagsList[] = {
+        0x0021, 0x0023, 0x0033, 0x0071, 0x0077, 0x006c, 0x0067, 0x007b, 0x02f2, 0x02f3,
+        0x1031, 0x1032, 0x13ab, 0x13ac, 0x15ee, 0x23a2, 0x23c6, 0x2e67, 0x33a4, 0x33c5,
+        0x3446, 0x2de7, 0x2df8, 0x26bf, 0x28ca, 0x3384, 0x13b8, 0x037e, 0x0485, 0x18d7,
+        0x0005, 0x0009, 0x0011, 0x0012, 0x0016, 0x0017, 0x0018, 0x0022, 0x0024, 0x0025,
+        0x0027, 0x002b, 0x002f, 0x003f, 0x004b, 0x004c, 0x004d, 0x004e, 0x004f, 0x006a,
+        0x006b, 0x006e, 0x007a, 0x007d, 0x0255, 0x3eb5, 0x3ea5, 0x3d7b, 0x33c4, 0x2fab,
+        0x2ebc, 0x29fc, 0x29a5, 0x2955, 0x2933, 0x135f, 0x2922, 0x26a5, 0x26fc, 0x2532,
+        0x23c9, 0x23c4, 0x23c5, 0x137f, 0x1378, 0x07e2, 0x07e3, 0x07e4, 0x0675, 0x05bc,
+        0x05bd, 0x05db, 0x05dd, 0x0598, 0x050d, 0x0444, 0x037c,
+
+        0x3314f, 0x43a770, 0x1eb923, 0x1cb923, 0xeb524, 0x1c83ab, 0x1e83bb
+    };
+
+    /*!
+      Tag Look-up list for Matroska Type Video Files
+      The Tags have been categorized in 4 categories. Which are
+      mentioned as a comment in front of them.
+      s   --  Tag to be Skipped
+      sd  --  Tag to be Skipped along with its data
+      u   --  Tag used directly for storing metadata
+      ui  --  Tag used only internally
+     */
+    extern const MatroskaTags matroskaTags[] = {
```

Comment:
extern const is an old way of doing things. This should be constexpr and probably std::array.

Reply:
done here https://github.com/Exiv2/exiv2/pull/2370/commits/65a01dc171f4dc230eed22754990d4c25dd0078d

Reply:
Hello @postscript-dev, thank you for your comment, It's really helpful, I will try to add unit test these days  ( install all dependencies and resolve them take time more the development Itself :) )
Hello @neheb , thank you for your notes. I appreciate that, I fixed the code and pushed a new commit .
Gays, please bear with, I'm new in this project and I'm not aware It's history, if you have further advises , I'm happy to take them.
Regards